lib/src/view/game/game_result_dialog.dart

```diff
@@ -89,6 +89,7 @@ class _GameEndDialogState extends ConsumerState<GameResultDialog> {
   Widget build(BuildContext context) {
     final ctrlProvider = gameControllerProvider(widget.id);
     final gameState = ref.watch(ctrlProvider).requireValue;
+    final ValueNotifier<bool> animationFinished = ValueNotifier(false);
```

Comment:
A `ValueNotifier` should not be declared in `build` method I think. If you do that you cannot properly `dispose` it.

Rather than using `ValueNotifier` and `ValueListenableBuilder` you can directly declare a `bool` in the state object and use `setState` in `onEnd`.